File: Excel/613 Challenge.py
import pandas as pd
from itertools import accumulate

path = "613 Sum of Digits Sequence.xlsx"
test = pd.read_excel(path, usecols="A", nrows=10001).squeeze()

def sum_of_digits(n):
    return sum(int(digit) for digit in str(n))

sequence = list(accumulate(range(1, 10001), lambda x, _: x + sum_of_digits(x), initial=1))
sequence = [1] + sequence[:-2]

print(all(sequence == test)) # True

File: Excel/613 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/613 Sum of Digits Sequence.xlsx"
test  = read_excel(path, range = "A1:A10001") %>% pull()

sum_of_digits <- function(n) {
  sum(as.numeric(strsplit(as.character(n), "")[[1]]))
}

sequence <- accumulate(1:10000, ~ .x + sum_of_digits(.x), .init = 1) %>%
  c(1, .) %>%
  head(-2)

all.equal(sequence, test)
#> [1] TRUE